docs/project-setup/filesystem-sync.md

```diff
@@ -0,0 +1,73 @@
+# Filesystem Sync with Unison
+
+The default NFS filesystems that Outrigger provides are very easy to setup and easy to share code and other files into
+your project containers, but NFS can be slow when writing, reading, or scanning thousands of files when compared to
+native filesystem performance. This can be a particular challenge when building your code base from a package manager onto
+a volume mounted NFS share.
+
+It also has the downside of not propagating filesystem modification events, which can be problematic if you wanted to have
+a process in a container monitor your files and run a process to rebuild or update when modification notifications are triggered.
+
+In order to better support these types of operations, Outrigger provides support for multi-directional filesystem syncing 
+(which includes filesystem notifications) called `unison`.  Your project root folder is sync'd via `unison` in to a sync 
+container and exposed as a Docker Volume to the rest of your application. This volume is then used as your mount point 
+instead of the NFS share and allows nearly native filesystem performance and features.
+
+## Sync Volume Name
+
+The name can be discovered in the following order
+
+1. Argument provided to the `project sync:start` command
+2. Specified in the `sync` -> `volume-name` property of your [Outrigger Project Configuration](./project-configuration.md)
+3. In your `docker-compose.yml` file, specified as an `external` volume with a name of the pattern `*-sync`
+4. If not specified anywhere else, it will use the name of the current project folder with `-sync` appended
+
+We recommend using approach #3, as it is the most explicit and is in line with all of the configuration you will need
+to take full advantage of `unison` syncing.
+
+## Setting up sync for your containers
+
+To setup `unison` sync for your containers you will need to [reference an external volume](https://docs.docker.com/compose/compose-file/#volume-configuration-reference) 
+in your `docker-compose.yml` (and in your `build.yml` where it really matters if you follow that convention) and in 
+the volume mount specification for the services that want to use it.
+
+**Add the volume spec to your compose file**
+
+```yaml
+volumes:
+  project-sync:
+    external: true
+```
+
+!!! important "Note about volume name"
+    The name in this `volumes` section must match the volume name you passed to the `project sync:start` command or the 
+    name specified in your Outrigger Project Configuration file, if you went that route. The reason why we don't exclusively
```

Comment:
If you optionally specified a volume name to the project sync:start command or configured a name in your Outrigger Project Configuration file the name in this `volumes` section must match the one specified.